src/osemgrep/networking/Rule_fetching.ml

```diff
@@ -156,9 +156,9 @@ type _registry_cached_value =
   Cache_disk.cached_value_on_disk
 
 (* better: faster fetching by using a cache *)
-let fetch_content_from_registry_url_async ~registry_caching url =
+let fetch_content_from_registry_url_async ~token_opt ~registry_caching url =
```

Comment:
Don't feel super strongly but might make sense to have this be optional since it is optional for `fetch_content_from_url{,_async}`.

Reply:
yeah – I'm leaning towards required since we had token passing "too optional" before and didn't pass the token prior.